src/ng/services/table.js

```diff
@@ -1,22 +1,143 @@
+class Dom {
+	column() {
+		return {cells: []};
+	}
+
+	row() {
+		return {cells: []};
+	}
+
+	cell() {
+		return null;
+	}
+
+	get view() {
+		return null;
+	}
+}
+class TableDom extends Dom {
+	constructor(element) {
+		super();
+		this.element = element;
+	}
+
+	static get empty() {
+		return new Dom();
+	}
+
+	get view() {
+		return this.element;
+	}
+
+	column(index) {
+		return new ColumnDom(this.element, index);
+	}
+
+	row(index) {
+		const rows = this.element.rows;
+		if (index >= 0 && index < rows.length) {
+			return new RowDom(rows[index]);
+		}
+		return null;
+	}
+
+	cell(row, column) {
+		const rows = this.element.rows;
+		row = row < 0 ? 0 : row;
+		column = column < 0 ? 0 : column;
+		const cellsCount = rows[0].cells.length;
+		if (row < rows.length && column < cellsCount) {
+			const cell = rows[row].cells[column];
+			return new CellDom(cell);
+		}
+		return null;
+	}
+}
+class CellDom extends Dom {
+	constructor(element) {
+		super();
+		this.element = element;
+	}
+
+	get view() {
+		return this.element;
+	}
+
+	get model() {
+		const scope = angular.element(this.element).scope();
+		if (scope) {
+			return scope.$cell;
+		}
+	}
+
+	addClass(name) {
+		this.element.classList.add(name);
+	}
+
+	removeClass(name) {
+		this.element.classList.remove(name);
+	}
+}
+
+class RowDom extends Dom {
+	constructor(element) {
+		super(element);
+		this.element = element;
+	}
+
+	get view() {
+		return this.element;
+	}
+
+	get cells() {
+		const row = this.element;
```

Comment:
I think we shouldn't make any property as getter, cause it will force user to make lake that
```javascript
if(table.body.row(0).cells.length){
table.body.row(0).cells.forEach
}
```
User won't see the perfomance impact of using `cells` 2 times